core/src/main/scala/org/bitcoins/core/wallet/builder/RawTxFinalizer.scala

```diff
@@ -280,6 +280,67 @@ object StandardNonInteractiveFinalizer {
   }
 }
 
+/** A finalizer which adds a change output, performs sanity checks,
+  * shuffles inputs and outputs, and adds non-signature witness data.
+  * This is the standard non-interactive finalizer within the Bitcoin-S wallet.
+  */
+case class ShufflingNonInteractiveFinalizer(
+    inputInfos: Vector[InputInfo],
+    feeRate: FeeUnit,
+    changeSPK: ScriptPubKey)
+    extends RawTxFinalizer {
+
+  override def buildTx(txBuilderResult: RawTxBuilderResult)(implicit
+      ec: ExecutionContext): Future[Transaction] = {
+    val addChange = ChangeFinalizer(inputInfos, feeRate, changeSPK)
+
+    val sanityCheck = SanityCheckFinalizer(
+      inputInfos = inputInfos,
+      expectedOutputSPKs = txBuilderResult.outputs.map(_.scriptPubKey),
+      expectedFeeRate = feeRate,
+      changeSPKs = Vector(changeSPK))
+
+    val addWitnessData = AddWitnessDataFinalizer(inputInfos)
+
+    addChange
+      .andThen(sanityCheck)
+      .andThen(addWitnessData)
+      .andThen(ShuffleFinalizer)
+      .buildTx(txBuilderResult)
+  }
+}
+
+object ShufflingNonInteractiveFinalizer {
```

Comment:
I think we should have a trait for this kind of `object` which has `txBuilderFrom` and `txFrom` with these same parameters.